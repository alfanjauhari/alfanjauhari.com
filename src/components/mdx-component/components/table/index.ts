export * from './table';
export * from './table.styles';
