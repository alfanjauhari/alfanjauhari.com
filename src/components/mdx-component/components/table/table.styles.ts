import { styled } from '@/theme';

export const StyledTable = styled('table', {
  my: '$5',
  width: '100%',
  tableLayout: 'fixed',
  borderSpacing: 0,
  '& thead': {
    '& th': {
      py: '$2',
      textAlign: 'start',
      fontSize: '$sm',
      borderBottom: '1px solid $colors$gray3',
      fontWeight: 'normal',
      color: '$gray6',
    },
    '& td': {
      py: '$2',
      borderBottom: '1px solid $colors$gray3',
      fontSize: '$sm',
    },
  },
  '& tbody': {
    width: '100%',
    '& tr': {
      '& td': {
        py: '$2',
        borderBottom: '1px solid $colors$gray3',
        fontSize: '$md',
      },
      '&:nth-child(odd)': {
        backgroundColor: '$gray1',
      },
    },
  },
});
