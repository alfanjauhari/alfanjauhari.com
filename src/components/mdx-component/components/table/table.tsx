import { TableHTMLAttributes } from 'react';
import { StyledTable } from './table.styles';

export type TableProps = TableHTMLAttributes<HTMLTableElement>;

export function Table(props: TableProps) {
  return <StyledTable {...props} />;
}
