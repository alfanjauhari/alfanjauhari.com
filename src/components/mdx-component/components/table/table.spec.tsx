import { render, screen } from '@testing-library/react';
import { Table } from './table';

describe('MDXComponent/Table', () => {
  it('should render successfuly', () => {
    render(
      <Table>
        <thead>
          <tr>
            <th>Column 1</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Data 1</td>
          </tr>
        </tbody>
      </Table>,
    );

    expect(screen.getByRole('table')).toBeVisible();
  });

  it('should match snapshot', () => {
    render(
      <Table>
        <thead>
          <tr>
            <th>Column 1</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>Data 1</td>
          </tr>
        </tbody>
      </Table>,
    );

    expect(screen.getByRole('table')).toMatchSnapshot();
  });
});
