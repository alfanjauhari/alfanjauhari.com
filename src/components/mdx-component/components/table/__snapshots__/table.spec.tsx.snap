// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MDXComponent/Table should match snapshot 1`] = `
<table
  class="c-eJrWME"
>
  <thead>
    <tr>
      <th>
        Column 1
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        Data 1
      </td>
    </tr>
  </tbody>
</table>
`;
